waddrmgr/migrations.go

```diff
@@ -254,3 +260,91 @@ func migrateRecursively(src, dst walletdb.ReadWriteBucket,
 	}
 	return nil
 }
+
+// populateBirthdayBlock is a migration that attempts to populate the birthday
+// block of the wallet. This is needed so that in the event that we need to
+// perform a rescan of the wallet, we can do so starting from this block, rather
+// than from the genesis block.
+//
+// NOTE: This migration cannot guarantee the correctness of the birthday block
+// being set as we do not store block timestamps, so a sanity check must be done
+// upon starting the wallet to ensure we do not potentially miss any relevant
+// events when rescanning.
+func populateBirthdayBlock(ns walletdb.ReadWriteBucket) error {
```

Comment:
with the sanity check, is this actually needed? If we set it to the current block, then the sanity check should decrease it down to the actual block.

Reply:
Yeah the migration is still needed because the sanity check can only run once the wallet has been initialized. Since we'll be making another migration to force a rescan from the birthday block, these migrations happen before the wallet's been initialized and will most likely happen back-to-back for most users, so we need to make sure the birthday block exists before the following migration.

Reply:
I might be missing something, but since we must handle the case where the estimate being way off (testnet), then shouldn't we be able to just set the birthday to the latest block we know, and let the sanity check "fix" this estimate?

Reply:
In the case of consecutive migrations that depend on this one, the sanity check won't be done until the wallet is initialized. Since we also have a rescan migration upcoming, the birthday block just needs to be a more sensible value than the genesis block. We'll be updating the `btcwallet` dependency within `lnd` that contains all of these migrations as part of one commit, so users will experience the migrations one after the other within the same database transaction before the wallet is initialized.